controllers/noderefutil/indexer.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package noderefutil implements NodeRef utils.
+package noderefutil
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// AddIndexToManager registers indexes with the cache.
+func AddIndexToManager(ctx context.Context, mgr ctrl.Manager) error {
```

Comment:
What about
```suggestion
func AddIndexesToManager(ctx context.Context, mgr ctrl.Manager) error {
```
(even do I prefer to drop this func given that it makes opaque which index we are adding to the manager